### PR TITLE
doc: release: add remaining release notes for TF-M, and list of new boards

### PR DESCRIPTION
**Pull request by ioannisg**
Add remaining release notes entries for TF-M for Zephyr
v2.6.0 release, mentioning the added samples and platforms.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

### CODE REVIEW

File: doc/releases/release-notes-2.6.rst [hunk 1]
```diff
@@ -376,6 +376,12 @@ Trusted Firmware-m
 * Configured QEMU to run Zephyr samples and tests in CI on mps2_an521_nonsecure
   (Cortex-M33 Non-Secure) with TF-M as the secure firmware component.
 * Synchronized Trusted-Firmware-M module to the upstream v1.3.0 release.
+* Added a new sample to run the PSA tests with Zephyr
+* Added a new sample to run the TF-m regression tests using the Zephyr build system
```
Comment (galak) [inline doc/releases/release-notes-2.6.rst:380]: Nit, `TF-m` -> `TF-M`

**Review by thedjnK — APPROVED** [review]

File: doc/releases/release-notes-2.6.rst (hunk 1 above)
Comment (ioannisg) [reply doc/releases/release-notes-2.6.rst]: fixed now

**Review by erwango — APPROVED** [review]
Reference to https://github.com/zephyrproject-rtos/zephyr/pull/34868 could be added once merged.

**Comment by ioannisg** [comment]
> Reference to #34868 could be added once merged.

Sure, and not only that - we might have to add some info about the mbedtls version we will be using in tfm. 
@erwango ok if we get this in for now, and revisit the release notes before the 2.6 release?

**Review by oyvindronningstad — APPROVED** [review]

**Pull request merged**